include/mbgl/style/layer.hpp

```diff
@@ -88,6 +89,8 @@ class Layer : public mbgl::util::noncopyable {
             return std::forward<V>(visitor)(*as<CustomLayer>());
         case LayerType::FillExtrusion:
             return std::forward<V>(visitor)(*as<FillExtrusionLayer>());
+        default:
```

Comment:
Makes sense. Although returning a `nullptr` is not an option here. I couldn't find a good way to return in all instantiations (eg. for `void`, stack based values and pointers), so I kept the runtime_error to appease gcc.